src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs

```
@@ -0,0 +1,290 @@
+            return EncryptDecryptHelper(securityContext, buffer, 0, size, headerSize, trailerSize, true,
+                out resultSize);
+        }
+
+        public SecurityStatus DecryptMessage(SafeDeleteContext securityContext, byte[] buffer, ref int offset, ref int count)
+        {
+            int resultSize;
+            SecurityStatus retVal = EncryptDecryptHelper(securityContext, buffer, offset, count, 0, 0, false, out resultSize);
+            if (SecurityStatus.OK == retVal)
+            {
+                count = resultSize;
+            }
+            return retVal;
+        }
+
+        public int QueryContextChannelBinding(SafeDeleteContext phContext, ChannelBindingKind attribute,
+            out SafeFreeContextBufferChannelBinding refHandle)
+        {
+            throw NotImplemented.ByDesignWithMessage(SR.net_MethodNotImplementedException);
+        }
+
+        public int QueryContextStreamSizes(SafeDeleteContext securityContext, out StreamSizes streamSizes)
+        {
+            streamSizes = null;
+            try
+            {
+                streamSizes = new StreamSizes(Interop.libssl.SslSizes.HEADER_SIZE, Interop.libssl.SslSizes.TRAILER_SIZE, Interop.libssl.SslSizes.SSL3_RT_MAX_PLAIN_LENGTH);
+                return 0;
+            }
+            catch
+            {              
+                return -1;
+            }
+        }
+
+        public int QueryContextConnectionInfo(SafeDeleteContext securityContext, out SslConnectionInfo connectionInfo)
+        {
+            bool gotReference = false;
+            connectionInfo = null;
+            try
+            {
+                securityContext.DangerousAddRef(ref gotReference);
+                Interop.libssl.SSL_CIPHER cipher = Interop.OpenSsl.GetConnectionInfo(securityContext.DangerousGetHandle());
+                connectionInfo =  new SslConnectionInfo(cipher);
+                return 0;
+            }
+            catch
+            {
+                return -1;
```

Why are we eating all exceptions?